crates/llms/src/chat/mod.rs

```diff
@@ -364,7 +367,36 @@ pub fn message_to_mistral(
 #[async_trait]
 pub trait Chat: Sync + Send {
     fn as_sql(&self) -> Option<&dyn SqlGeneration>;
-    async fn run(&self, prompt: String) -> Result<Option<String>>;
+    async fn run(&self, prompt: String) -> Result<Option<String>> {
+        let span = tracing::Span::current();
+
+        async move {
```

Comment:
Standardise between OpenAI and Anthropic. Local models have their own implementation.